File: cron/stats_reports.py
from aiogram import Bot
import asyncio
from src.bot import config, db


async def main():
    bot = Bot(config.BOT_TOKEN)
    conn = db.get_connection(config.DB)
    with conn.cursor() as cur:
        cur.execute(
            "SELECT employee_id, SUM(total_hours) AS hours FROM work_sessions "
            "WHERE date >= CURDATE() - INTERVAL 7 DAY GROUP BY employee_id"
        )
        stats = cur.fetchall()
    for row in stats:
        with conn.cursor() as c2:
            c2.execute('SELECT telegram_id FROM employees WHERE id=%s', (row['employee_id'],))
            user = c2.fetchone()
            if user:
                await bot.send_message(user['telegram_id'], f"Ваши отработанные часы за неделю: {row['hours']}")
    conn.close()


if __name__ == '__main__':
    asyncio.run(main())

File: cron/stats_reports.php
<?php
require __DIR__ . '/../vendor/autoload.php';
$config = file_exists(__DIR__ . '/../src/bot/config.php')
    ? require __DIR__ . '/../src/bot/config.php'
    : require __DIR__ . '/../src/bot/config.sample.php';

use TelegramBot\Api\BotApi;

$bot = new BotApi($config['bot_token']);
$db = new PDO(
    "mysql:host={$config['db']['host']};dbname={$config['db']['database']};charset={$config['db']['charset']}",
    $config['db']['user'],
    $config['db']['password']
);

// Send weekly statistics (sum of hours for last 7 days)
$stmt = $db->query("SELECT employee_id, SUM(total_hours) AS hours FROM work_sessions WHERE date >= CURDATE() - INTERVAL 7 DAY GROUP BY employee_id");
$stats = $stmt->fetchAll(PDO::FETCH_ASSOC);

foreach ($stats as $row) {
    $emp = $db->prepare('SELECT telegram_id FROM employees WHERE id=?');
    $emp->execute([$row['employee_id']]);
    $tg = $emp->fetchColumn();
    if ($tg) {
        $bot->sendMessage($tg, "Ваши отработанные часы за неделю: {$row['hours']}");
    }
}

// Extend to monthly statistics as needed
